src/DcmTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,652 @@
+  double doubleSupportStartTime;
+  getFirstDoubleSupportTiming(doubleSupportStartTime);
+
+  if(!addFirstDoubleSupportPhase(doubleSupportStartTime,
+  				 initPosition,
+  				 initVelocity)){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the DCM trajectory of the first DS phase is generated." << std::endl;
+    return false;
+  }
+
+  // evaluate the DCM trajectory
+  if(!evaluateDcmTrajectory()){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the whole DCM trajectory is evaluated." << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+bool DcmTrajectoryGenerator::evaluateDcmTrajectory()
+{
+  size_t timeVectorLength = std::get<1>(m_trajectoryDomain) - std::get<0>(m_trajectoryDomain);
+  std::vector<size_t> timeVector(timeVectorLength);
+  
+  // populate the time vector
+  std::iota(std::begin(timeVector), std::end(timeVector), std::get<0>(m_trajectoryDomain));
+
+  // clear all the previous DCM position 
+  m_dcmPos.clear();
+  m_dcmPos.reserve(timeVectorLength);
+
+  // clear all the previous DCM velocity 
+  m_dcmVel.clear();
+  m_dcmVel.reserve(timeVectorLength);
+
+  iDynTree::Vector2 dcmPos, dcmVel;
+  double time;
+  std::vector<std::shared_ptr<GeneralSupportTrajectory>>::reverse_iterator subTrajectory = m_trajectory.rbegin();
+  
+  for (size_t t : timeVector){
+    time = t * m_dT;
+    double subTrajectoryEndTime  = std::get<1>((*subTrajectory)->getTrajectoryDomain());
+    if (time > subTrajectoryEndTime){
+      subTrajectory++;
+    }
+    
+    if(!(*subTrajectory)->getDcmPos(time, dcmPos)){
+      std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the position of the DCM is evaluated." << std::endl;
+      return false;
+    }
+
+    if(!(*subTrajectory)->getDcmVel(time, dcmVel)){
+      std::cerr << "[DCM TRAJECTORY GENERATOR] Error when the velocity of the DCM is evaluated." << std::endl;
+      return false;
+    }
+    m_dcmPos.push_back(dcmPos);
+    m_dcmVel.push_back(dcmVel);
+  }
+  return true;
+}
+
+const std::vector<iDynTree::Vector2>& DcmTrajectoryGenerator::getDcmPosition() const
+{
+  return m_dcmPos;
+}
+
+const std::vector<iDynTree::Vector2>& DcmTrajectoryGenerator::getDcmVelocity() const
+{
+  return m_dcmVel;
+}
+
+bool DcmTrajectoryGenerator::setPauseConditions(const double &maxDoubleSupportDuration, const double &nominalDoubleSupportDuration)
+{
+  if (maxDoubleSupportDuration < 0){
+    std::cerr << "[DCM TRAJECTORY GENERATOR] If the maxDoubleSupportDuration  is negative, the robot won't pause in middle stance." << std::endl;
+    m_pauseActive = false;
+  }
+  
+  m_pauseActive = true;
+  m_maxDoubleSupportDuration = maxDoubleSupportDuration;
+  
+  if (m_pauseActive){
+    if (nominalDoubleSupportDuration <= 0){
+      std::cerr << "[DCM TRAJECTORY GENERATOR] The nominalDoubleSupportDuration is supposed to be positive." << std::endl;
+      m_pauseActive = false;
+      return false;
+    }
+    
+    if (nominalDoubleSupportDuration > maxDoubleSupportDuration / 2){
```

Comment:
Why ``/2``? Also it is not consistent with the error message.

Reply:
Yes the error message is not consistent. I made a mistake. I try to explain the why `/2` is wrong.   
Suppose that the DS duration is equal to `maxDoubleSupportDuration`. In this case (if `m_pauseActive == true`) the robot has to stop in the center of the support polygon for a period equals to  `nominalDoubleSupportDuration - maxDoubleSupportDuration` thus the `if` statement has to become 
```c++
if (nominalDoubleSupportDuration > maxDoubleSupportDuration){
   ....
}
```